src/core/utils/string.cpp

```diff
@@ -106,6 +106,77 @@ bool Utils::String::naturalSort(QString left, QString right, bool &result)
     return false;
 }
 
+bool Utils::String::naturalLessThan(const QString &left, const QString &right)
+{
+    // Return value `false` indicates QString `right` should go before QString `left`, otherwise, after
+    const bool case_sensitive = false;
+    int posL = 0;
+    int posR = 0;
+    while (true) {
+        while (true) {
+            if (posL == left.size() || posR == right.size())
+                return (left.size() < right.size());  // when a shorter string is another string's prefix, shorter string place before longer string
+
+            QChar leftChar = left[posL];
+            QChar rightChar = right[posR];
+            if (leftChar.isDigit() && rightChar.isDigit())
+                break; // Both are digits, break this loop and compare numbers
+            else if (leftChar.isLetter() && rightChar.isLetter()) {
+                if(!case_sensitive) {
+                    // special case for case insensitive comparsion
+                    QChar l = leftChar.toLower();
+                    QChar r = rightChar.toLower();
+                    if(l != r)
+                        return (l < r);
+                }
+
+                // case sensitive comparsion
+                if (leftChar != rightChar) {
+                    if (leftChar.isUpper() && !rightChar.isUpper())
+                        return false;
+                    else if (!leftChar.isUpper() && rightChar.isUpper())
+                        return true;
+                    else
+                        return (leftChar < rightChar);
+                }
+            }
+            else
```

Comment:
At this point the 2 chars might be equal and you should proceed to compare the next ones with out exiting.
I propose, after getting the 2 chars at the top you should do:
```c++
if (leftChar == rightChar)
    continue;
```
This would simplify the some if tests too.